File: typescript/debug.ts
// debug.ts
import { PolisClient } from "./src/polis_client";

async function main() {
  console.log("Creating an anonymous client…");
  const polis = new PolisClient();

  try {
    console.log("Fetching comments…");
    const comments = await polis.getComments("2demo");
    // For less data returned, like in raw API response:
    // const comments = await polis.getComments("2demo", { moderation: false, include_voting_patterns: false } );
    console.log("Comments count:", comments?.length);
    console.log("Sample comment:", comments?.[0]);

    console.log("\nFetching report…");
    const report = await polis.getReport("r68fknmmmyhdpi3sh4ctc");
    console.log("Report:", report);

    console.log("\nFetching conversation…");
    const convo = await polis.getConversation("2demo");
    console.log("Conversation:", convo);

    console.log("\nFetching math…");
    const math = await polis.getMath("2demo");
    console.log("Math:", math);

    console.log("\nFetching participationInit…");
    const init = await polis.getInitialization("2demo");
    console.log("ParticipationInit:", init);

  } catch (err) {
    console.error("\n❌ Error during anon debug run:");
    console.error(err);
    // @ts-expect-error
    process.exit(1);
  }

  console.log("Creating an authenticated xid client…");
  const polisAuth = new PolisClient({ xid: "foobar" });

  try {
    console.log("Fetching an auth token…");
    await polisAuth.fetchToken("2demo");

    console.log("\nFetching conversation uuid…");
    const conversationUuid = await polisAuth.getConversationUuid("2demo") ?? {} ;
    console.log(conversationUuid);

  } catch (err) {
    console.error("\n❌ Error during anon debug run:");
    console.error(err);
    // @ts-expect-error
    process.exit(1);
  }
}

main();

File: python/debug.py
from polis_client import PolisClient

polis = PolisClient()

comments = polis.get_comments(conversation_id="2demo")
if isinstance(comments, list) and len(comments) > 0:
    print(comments[0].to_dict())

report = polis.get_report(report_id="r68fknmmmyhdpi3sh4ctc")
if report:
    print(report.to_dict())

conversation = polis.get_conversation(conversation_id="2demo")
if conversation and not isinstance(conversation, str):
    print(conversation.to_dict())

math = polis.get_math(conversation_id="2demo")
print(math)

votes = polis.get_votes(conversation_id="2demo", pid=10)
if isinstance(votes, list) and len(votes) > 0:
    print(votes[0].to_dict())

polis_auth = PolisClient(xid="foobar")
polis_auth._ensure_token(conversation_id="2demo")
xids = polis_auth.get_conversation_xids(conversation_id="2demo")
if xids:
    print(xids)